### PR TITLE
refactor(data-structures): rename "version" field to "signals" in BlockHeader

### PR DESCRIPTION
**Pull request by lrubiorod**

### CODE REVIEW

File: data_structures/src/chain.rs [hunk 1]
```diff
@@ -636,7 +636,8 @@ impl Hashable for PublicKey {
 #[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize, ProtobufConvert, Default)]
 #[protobuf_convert(pb = "witnet::Block_BlockHeader")]
 pub struct BlockHeader {
-    /// The block version number indicating the block validation rules
+    /// 32 bits for binary signaling new witnet protocol improvements
+    // TODO: Rename `version` to `signals` when we support migrations
```
Comment (aesedepece) [inline data_structures/src/chain.rs:640]: Why? AFAIK fields can be safely renamed.

File: data_structures/src/chain.rs [hunk 2]
```diff
@@ -636,7 +636,8 @@ impl Hashable for PublicKey {
...
     pub version: u32,
```
Comment (tmpolaczyk) [inline data_structures/src/chain.rs:641]: Renaming is supported by both bincode and protobuf, but it would break the JSON-RPC API.

**Comment by tmpolaczyk** [comment]
So this definitely breaks the JSON-RPC API, but in an expected way. We assume that nobody was using the old `version` field, and any client that wants to add a signaling monitor will need to update in order to support it.

However, this will break any client that uses the `getBlock` method, for example the wallet. This is because the version field is needed to deserialize the JSON object as a witnet `Block`. This wouldn't be a problem if the `getBlock` method used protobuf as the serialization format. Possible solutions:

1. Do nothing, close this PR
2. Use `serde(rename = "version")` to keep compatibility in the JSON-RPC API
3. Add an extra version field after the serialization, similar to how we add the "blockHashes" field.
4. Break the API. This means updating all the nodes and all the clients that may be affected. We may want to use this opportunity to change other aspects of the JSON-RPC API.

I prefer 1) because it's the one with less unexpected side effects.

**Comment by mariocao** [comment]
Option 4 is mentioned in the [WIP-0014](https://github.com/witnet/WIPs/blob/master/wip-0014.md) as the improvement labeled as **(1)**.

**Review by mariocao — COMMENTED** [review]
Option 4 is mentioned in the [WIP-0014](https://github.com/witnet/WIPs/blob/master/wip-0014.md) as the improvement labeled as (1).

**Review by aesedepece — APPROVED** [review]

File: data_structures/src/chain.rs [hunk 3]
```diff
@@ -638,8 +638,8 @@ impl Hashable for PublicKey {
 #[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize, ProtobufConvert, Default)]
 #[protobuf_convert(pb = "witnet::Block_BlockHeader")]
 pub struct BlockHeader {
-    /// The block version number indicating the block validation rules
-    pub version: u32,
+    /// 32 bits for binary signaling new witnet protocol improvements
```
Comment (aesedepece) [inline data_structures/src/chain.rs:641]: Let's link to WIP-0014 here

**Review by aesedepece — CHANGES_REQUESTED** [review]

File: data_structures/src/chain.rs (hunk 3 above)
Comment (mariocao) [reply data_structures/src/chain.rs:641]: That would be great

File: data_structures/src/chain.rs [hunk 4]
```diff
@@ -638,8 +638,9 @@ impl Hashable for PublicKey {
...
+    /// https://github.com/witnet/WIPs/blob/master/wip-0014.md
```
Comment (aesedepece) [inline data_structures/src/chain.rs:642]:
```suggestion
    /// 32 bits for binary signaling new witnet protocol improvements.
    /// See [WIP-0014](https://github.com/witnet/WIPs/blob/master/wip-0014.md) for more info.
```

**Review by aesedepece — CHANGES_REQUESTED** [review]

File: validations/src/tests.rs [hunk 5]
```diff
@@ -7521,7 +7521,7 @@ fn block_change_version() {
     // invalidate the signature
     let x = test_block(|b| {
         // Flip one bit of the version field
```
Comment (tmpolaczyk) [inline validations/src/tests.rs:7523]:
```suggestion
        // Flip one bit of the signals field
```

File: validations/src/tests.rs [hunk 6]
```diff
@@ -7521,7 +7521,7 @@ fn block_change_version() {
...
-        b.block_header.version ^= 1;
+        b.block_header.signals ^= 1;
 
         // Do not sign the block again
         // If the miner changes the version field and then signs the block, the block will be valid
```
Comment (tmpolaczyk) [inline validations/src/tests.rs:7527]:
```suggestion
        // If the miner changes the signals field and then signs the block, the block will be valid
```

File: validations/src/tests.rs (hunk 5 above)
Comment (tmpolaczyk) [reply validations/src/tests.rs:7523]: And also the test name

**Review by tmpolaczyk — APPROVED** [review]

**Review by aesedepece — APPROVED** [review]

**Pull request merged**